relay/pkg/btc/remote/remote.go

```diff
@@ -11,19 +16,83 @@ var logger = log.Logger("relay-btc-remote")
 
 // remoteChain represents a remote Bitcoin chain.
 type remoteChain struct {
-	// TODO: implementation
+	rpcClient *rpcclient.Client
+}
+
+func serializeHeader(header *wire.BlockHeader) ([]byte, error) {
+	var buffer bytes.Buffer
+	writer := io.Writer(&buffer)
+	err := header.Serialize(writer)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return buffer.Bytes(), nil
 }
 
 // Connect connects to the Bitcoin chain and returns a chain handle.
-func Connect() (btc.Handle, error) {
-	// TODO: implementation
-	logger.Infof("connecting remote Bitcoin chain")
+func Connect(config *btc.Config) (btc.Handle, error) {
+	connCfg := &rpcclient.ConnConfig{
+		User:         config.Username,
+		Pass:         config.Password,
+		Host:         config.URL,
+		HTTPPostMode: true, // Bitcoin core only supports HTTP POST mode
+		DisableTLS:   true, // Bitcoin core does not provide TLS by default
+	}
+
+	client, err := rpcclient.New(connCfg, nil)
+
```

Comment:
In general, the `if err != nil` is directly under the statement which produces that `err`. No blank lines needed.